Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/NewMeterWizard/VirtualChannelModal.tsx

```diff
@@ -0,0 +1,317 @@
+    }
+
+    function handleVCModalConfirmCallback(conf, isButton) {
+        const mTypeArr = [];
+        const phaseArr = [];
+        for (const ch of virtualChannels) {
+            mTypeArr.push(ch.MeasurementType);
+            phaseArr.push(ch.Phase);
+        }
+
+        if (conf) {
+            let channel: OpenXDA.Types.Channel = {
+                ID: 0, // gets overwritten by addChannel in ChannelPage
+                Meter: props.MeterKey, Asset: '',
+                MeasurementType: mTypeArr.every(v => v === mTypeArr[0]) ? mTypeArr[0] : 'Voltage',   // if all the same return one of the el's
+                Adder: 0, Phase: phaseArr.every(v => v === phaseArr[0]) ? phaseArr[0] : 'AN', 
+                MeasurementCharacteristic: 'Instantaneous',
+                Name: 'New Virtual Channel',  Multiplier: 1,
+                SamplesPerHour: 0, PerUnitValue: null,
+                HarmonicGroup: 0, Description: 'Voltage AN',
+                Enabled: true, Series: [{
+                    ID: 0,
+                    ChannelID: 0,
+                    SeriesType: 'Values',
+                    SourceIndexes: virtualChannels.map((ch) => ch.Scale + '*' + ch.Series).join(',') // scal * source idx
+                } as OpenXDA.Types.Series],
+                ConnectionPriority: 0, Trend: false
+            } as OpenXDA.Types.Channel
+
+            let channels: Array<OpenXDA.Types.Channel> = [channel].concat(_.cloneDeep(props.Channels));
+            props.UpdateChannels(channels);
+
+            props.Close(channel)
+            setVirtualChannels([]);
+        } else if (isButton) {
+            props.Close(undefined);
+            setVirtualChannels([]);
+        }
+    }
+
+    function updateChannelScale(vChannel: IVirtualChannel, index: number) {
+        let updatedChannels = _.cloneDeep(virtualChannels);
+
+        if (index > -1) updatedChannels[index] = vChannel;
+        else updatedChannels.push(vChannel);
+
+        setVirtualChannels(updatedChannels);
+    }
+
+    function signedScale(index: number) {
+        const vChannels = _.cloneDeep(virtualChannels);
+
+        if (index > -1) vChannels[index].Scale = -vChannels[index].Scale;
+        setVirtualChannels(vChannels);
+    }
+
+    function removeVC(channel: IVirtualChannel) {
+        setVirtualChannels(prevChannels => [...prevChannels.filter((element) => element !== channel)]);
+    }
+
+    return <>
+        <Modal
+            Title={'Virtual Channel'}
+            Show={props.Shown}
+            CallBack={(conf, isButton) => {
+                handleVCModalConfirmCallback(conf, isButton);
+            }}
+            ConfirmText={'Add'}
+            ConfirmBtnClass={'btn-primary'}
+            DisableConfirm={disableModalConfirm}
+            ShowCancel={true}
+            Size={'lg'}
+            ConfirmShowToolTip={!validSelection}
+            ConfirmToolTipContent={<p><span>{CrossMark}</span>Selection includes channel with no Series.</p>}
```

Comment:
Should this be `Identifier` instead of `Series` since we're changing the labels?